File: src/main/java/io/github/starwishsama/namelessbot/objects/group/GroupConfig.java
package io.github.starwishsama.namelessbot.objects.group;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.LinkedList;
import java.util.List;

@Data
public class GroupConfig {
    @SerializedName("group_id")
    private Long groupId;
    @SerializedName("auto_accept")
    private boolean autoAccept;
    private List<Long> admins = new LinkedList<>();
    @SerializedName("mc_server_info")
    private boolean mcServerInfo;
    @SerializedName("mc_server_address")
    private String mcServerAddress;

    public GroupConfig(Long id){
        groupId = id;
    }
}

File: src/main/kotlin/io/github/starwishsama/nbot/objects/group/GroupConfig.kt
package io.github.starwishsama.nbot.objects.group

import com.google.gson.annotations.SerializedName
import java.util.*

class GroupConfig(@SerializedName("group_id") val groupId: Long) {

    @SerializedName("auto_accept")
    val autoAccept = false

    val admins: List<Long> = LinkedList()

    @SerializedName("mc_server_info")
    val mcServerInfo = false

    @SerializedName("mc_server_address")
    val mcServerAddress: String? = null

}